src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -914,5 +983,32 @@ private static string CreateDeterminismKey(CommandLineArguments args, string[] r
 
             return builder.ToString();
         }
+
+
+        /// <summary>
+        /// The set of source file paths that are in the set of embedded paths.
+        /// This is used to prevent reading source files that are embedded twice.
+        ///
+        /// It further prevents a race where the file we embed could have changed
+        /// since the time it was parsed.
+        /// </summary>
+        protected IReadOnlySet<string> EmbeddedSourcePaths { get; }
```

Comment:
Moved. It is used but IReadOnlySet has only Count and Contains, we never enumerate it and so order does not matter.